custom_components/google_home/api.py

```diff
@@ -227,3 +229,89 @@ async def update_google_devices_information(self) -> List[GoogleHomeDevice]:
             ]
         )
         return coordinator_data
+
+    async def delete_timer_or_alarm(
+        self, device: GoogleHomeDevice, item_to_delete: str
+    ) -> None:
+        """Deletes a timer or alarm,
+        can also delete multiple if a list is provided (Not implemented yet)."""
+
+        url = self.create_url(str(device.ip_address), PORT, API_ENDPOINT_DELETE)
```

Comment:
Check the value here. Same for `device.auth_token`.
```suggestion
        if device.ip_address is None:
           _LOGGER.error("Device %s doesn't have IP address!", device.name)
           return
        url = self.create_url(device.ip_address, PORT, API_ENDPOINT_DELETE)
```

Reply:
I am not checking it. But when i don't convert it to string it throws and error saying `'Optional[str]'; got 'str'`. Do you know why?

Reply:
Because it may not be set. By casting to str, you're hiding this error, and ip may be set to "None" (str).

Reply:
I see 😄 I didn't know that `mypy` is so intelligent that it can check these things. This is so awesome!!